package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -1084,7 +1096,12 @@ def _getMeanOnePoint(self,universe,selection1,selection_str,dt,totalFrames):
             valOList.append(a)
             sumsdt +=  dt
             n += 1
-        return sumDeltaO / n
+            
+        #if no water molecules remain in selection, there is nothing to get the mean, so n = 0.
+        if n == 0:
+            return (0,0,0)
```

Comment:
You're right, my mistake, next time I'll implement tests for this "little" changes.